pipeline/full_pipeline.py

```diff
@@ -0,0 +1,230 @@
+    parser.add_argument(
+        "--interpolator-name",
+        type=str,
+        choices=("rife", "cain", "maskflownet", "raftnet"),
+        help="""\
+        Name of the interpolator model.
+        """,
+    )
+    parser.add_argument(
+        "--interpolator-checkpoint",
+        type=str,
+        help="""\
+        Path of the interpolator checkpoints.
+        """,
+    )
+    parser.add_argument(
+        "--reference-path",
+        type=Path,
+        help="""\
+        Path to the reference volume used for the optical flow prediction.
+        If interpolation model is "cain" or "rife", there is no need to 
+        specify a reference path.
+        """,
+    )
+    parser.add_argument(
+        "--output-dir",
+        type=Path,
+        help="""\
+        Path to directory where to save the results.
+        """,
+    )
+    parser.add_argument(
+        "-f",
+        "--force",
+        action="store_true",
+        help="""\
+        If True, force to recompute every steps.
+        """,
+    )
+    return parser.parse_args()
+
+
+def get_experiments_list_to_run(
+    gene_name: str,
+    experiment_path: Path | str,
+    force: bool,
+) -> set[int]:
+    """Returns the set of experiments still to be runned.
+
+    Parameters
+    ----------
+    gene_name
+        Name of the gene to experiment.
+    experiment_path
+        Path where the results of the given experiment are saved.
+    force
+        If force is True, all the experiments are going to be returned
+        even if some paths already exist.
+
+    Returns
+    -------
+    experiments_list: set[int]
+        Set of experiments IDs to run.
+    """
+    experiments_ids = get_experiment_list_from_gene(gene_name, "coronal")
+    experiments_ids.extend(get_experiment_list_from_gene(gene_name, "sagittal"))
+
+    if not Path(experiment_path).exists() or force:
+        return set(experiments_ids)
+
+    experiments_results = {
+        int(filename.stem.split("-")[0]) for filename in experiment_path.iterdir()
+    }
+    experiments_list = set(experiments_ids) - experiments_results
+    return experiments_list
+
+
+def main(
+    nissl_path: Path | str,
+    ccfv2_path: Path | str,
+    ccfv3_path: Path | str,
+    gene_name: str,
+    interpolator_name: str,
+    interpolator_checkpoint: Path | str,
+    output_dir: Path | str | None = None,
+    reference_path: str | None = None,
+    force: bool = False,
+) -> int:
+    """Implement the main function."""
+    from download_gene import main as download_gene_main
+    from gene_to_nissl import main as gene_to_nissl_main
+    from interpolate_gene import main as interpolate_gene_main
+    from nissl_to_ccfv3 import main as nissl_to_ccfv3_main
+
+    if output_dir is None:
+        output_dir = Path(__file__).parent / "full-pipeline"
+    else:
+        output_dir = Path(output_dir)
+
+    warped_nissl_path = output_dir / "nissl-to-ccfv3" / "warped-nissl.npy"
+    if not warped_nissl_path.exists() or force:
```

Comment:
I guess this is one of the reasons why you were hardcoding paths in the independent subscripts.

I guess your approach is similar to what luigi does - by default it considers a specific stage of a pipeline to be done if a specific folder exists.

IMO we can implement similar logic purely via the CLI by using mutually exclusive groups (https://docs.python.org/3/library/argparse.html#mutual-exclusion). What do I mean by it? Well, in many cases we either have a result of a stage lying on the disk and we just want to provide its path. However, sometimes we don't have it and we want to run the stage and provide its input arguments. And the goal is to force the user to provide exactly one of them.

However, this is something to be disucssed.

Reply:
Whatever